src/lib/store.js

```diff
@@ -11,11 +11,20 @@ import thunk from 'redux-thunk';
 import rootReducer from './state/reducers';
 import { INITIAL_STORE_STATE } from './storeConfig';
 
+const preloadFiles = (files) =>
+  files
+    .map((file) => ({ state: 'finished', ...file }))
+    .reduce((acc, current) => {
+      acc[current.filename] = { ...current };
```

Comment:
I am not sure what you are trying to do here: isn't another `map` rather than a `reduce`?

Reply:
`files` coming from backend are a list and with the `reduce` I am converting them into an object with key the filename and value the object found in the list. E.g

```javascript
// Backend files
files = [{filename: 'Foo.txt', size: 1000, ...}

// Redux files strucutre
files = {
  'Foo.txt': {
    size: 1000,
    ....
  }
}

```